src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ClickHouseSinkBuffer.java

```diff
@@ -23,7 +23,7 @@
     private final List<String> localValues;
     private final List<CompletableFuture<Boolean>> futures;
 
-    private volatile long lastAddTimeMillis = 0L;
+    private volatile long lastAddTimeMillis = System.currentTimeMillis();
```

Comment:
why do we need to set up it here?

Reply:
Do you mean `System.currentTimeMillis()` instead `0L`?

Reply:
yes

Reply:
I removed `if (lastAddTimeMillis == 0)` condition in `checkTime()` to optimize.
If take `0L` in current code version, than `checkTime()` gives true at first check `flushCondition()`, but it shouldn't.

Reply:
ok